src/main/java/edu/cornell/kfs/coa/batch/service/impl/CreateClosedAccountsCsvServiceImpl.java

```diff
@@ -0,0 +1,303 @@
+        DateTime datePreviousAsDateTime = todayAsDateTime.minusDays(daysPrevious);
+        LOG.info("computeDateBeingDaysPreviousFromToday: datePreviousAsDateTime = " + datePreviousAsDateTime.toLocalDate().toString());
+        return datePreviousAsDateTime.toLocalDate();
+    }
+    
+    private void writeClosedAccountsToCsvFormattedFile(Map<String, Date> fileDataRangeSpecified) throws IOException {
+        LOG.info("writeClosedAccountsToCsvFormattedFile: Prior to ensuring directories to hold outbound file exist.");
+        ensureOutbundDirectoriesExist();
+        
+        LOG.info("writeClosedAccountsToCsvFormattedFile: Prior to obtaining data with SQL query for date range. This may take a while.");
+        List<ClosedAccount> closedAccountsData = getClosedAccountsByDateRangeDao().obtainClosedAccountsDataFor(fileDataRangeSpecified);
+        
+        if (ObjectUtils.isNotNull(closedAccountsData) && closedAccountsData.size() > 0) {
+            LOG.info("writeClosedAccountsToCsvFormattedFile: Have the data. Will now write out " + closedAccountsData.size() + " closed accounts to CSV file.");
+            writeClosedAccountsToCsvFile(closedAccountsData);
+        } else {
+            LOG.info("writeClosedAccountsToCsvFormattedFile: NO CSV file being created. " 
+                    + (ObjectUtils.isNotNull(closedAccountsData) ? (closedAccountsData.size() + " accounts being written to file.") : ("null returned for requested account data.") ));
+        }
+    }
+    
+    private void ensureOutbundDirectoriesExist() {
+        try {
+            FileUtils.forceMkdir(new File(getCsvClosedAccountstFileCreationDirectory()));
+        } catch (IOException e) {
+            LOG.info("ensureOutbundDirectoriesExist: Could not make file creation directory => " + getCsvClosedAccountstFileCreationDirectory() + " Throwing RuntimeException to force batch job failure.");
+            throw new RuntimeException(e);
+        }
+        try {
+            FileUtils.forceMkdir(new File(getCsvClosedAccountsExportDirectory()));
+        } catch (IOException e) {
+            LOG.info("ensureOutbundDirectoriesExist: Could not make file export directory => " + getCsvClosedAccountsExportDirectory() + " Throwing RuntimeException to force batch job failure.");
+            throw new RuntimeException(e);
+        }
+    }
+    
+    private void writeClosedAccountsToCsvFile(List<ClosedAccount> closedAccountsDataList) throws IOException {
+        String csvFileName = generateCsvOutputFileName();
+        String fullyQualifiedCreationDirectoryFileName = fullyQualifyFileNameToCreationDirectory(csvFileName);
+        LOG.info("writeClosedAccountsToCsvFile: fullyQualifiedOutputFile = " + fullyQualifiedCreationDirectoryFileName);
+        File outputFile = new File(fullyQualifiedCreationDirectoryFileName);
+        FileWriter outputFileWriter;
+        outputFileWriter = new FileWriter(outputFile);
```

Comment:
These could be consolidated into one line.

Reply:
Although these lines didn't get consolidated, this setup is okay as-is and the code can be moved forward without the cleanup.